### PR TITLE
Use more compact cache representation for int and str

### PR DESCRIPTION
**Pull request by ilevkivskyi**
After looking more at some real data I found that:
* More than 99.9% of all `int`s are between -10 and 117. Values are a bit arbitrary TBH, the idea is that we should include small negative values (for `TypeVarId`s) and still be able to fit them in 1 byte.
* More than 99.9% of strings are shorter that 128 bytes (again the idea is to fit the length into a single byte)

Note there are very few integers that would fit in two bytes currently. This is because we only store line for type alias nodes, and type aliases are usually defined at the top of a module. We can add special case for two bytes later when needed.

We could probably save another byte for long strings and medium integers, but I don't want to have anything fancy that would only affect less than 0.1% cases.

Finally you may notice I add a small correctness change I noticed accidentally when working on this, it is not really related, but it is so minor that it doesn't deserve a separate PR.

### CODE REVIEW

File: mypyc/test-data/run-classes.test [hunk 1]
```diff
@@ -2741,7 +2741,9 @@ def test_buffer_roundtrip() -> None:
     write_tag(b, TAG_B)
     write_int(b, 2)
     write_int(b, 2 ** 85)
+    write_int(b, 255)
     write_int(b, -1)
+    write_int(b, -255)
```
Comment (JukkaL) [inline mypyc/test-data/run-classes.test:2746]: Test also the edge cases (-11, -10, -9, 116, 117, 118). Test a few more different lengths of integers (e.g. 15 bits, 23 bits, 30 bits) with arbitrary lower bits.

**Review by JukkaL — COMMENTED** [review]
Nice -- did you check how much this reduces the size of binary cache files?

**Comment by ilevkivskyi** [comment]
@JukkaL
> did you check how much this reduces the size of binary cache files?

It looks like it's ~40% smaller now, for example:
* `.mypy_cache/3.12/types.data.ff 183K (master)`
* `.mypy_cache/3.12/types.data.ff 103K (this PR)`
* also btw `.mypy_cache/3.12/types.data.json 381K`

File: mypyc/test-data/run-classes.test (hunk 1 above)
Comment (ilevkivskyi) [reply mypyc/test-data/run-classes.test:2746]: OK, I think it would also make sense to test something like `len(data.getvalue()) == 1` etc.
